Sim/SimG4Components/src/G4ConstantMagneticFieldTool.cpp

```diff
@@ -1,137 +1,117 @@
-  	declareProperty( "MaximumEpsilon" , m_maxEps , "Maximum epsilon (see G4 documentation)" );
-  	declareProperty( "MaximumStep" , m_maxStep , "Maximum step length in field (see G4 documentation)" );
-
-  	declareProperty( "IntegratorStepper" , m_integratorStepper="NystromRK4","Integrator stepper name" );
-	declareProperty( "FieldComponentX",m_fieldComponentX,"Field X component");
-	declareProperty( "FieldComponentY",m_fieldComponentY,"Field Y component");
-	declareProperty( "FieldComponentZ",m_fieldComponentZ,"Field Z component");
-	declareProperty( "FieldRMax",m_fieldRadMax,"Field max radius");
-	declareProperty( "FieldZMax",m_fieldZMax,"field max Z");
+DECLARE_COMPONENT(G4ConstantMagneticFieldTool)
+
+G4ConstantMagneticFieldTool::G4ConstantMagneticFieldTool(const std::string& type, const std::string& name,
+                                                         const IInterface* parent)
+    : GaudiTool(type, name, parent),
+      m_field(nullptr),
+      m_fieldOn(false),
+      m_minEps(0),
+      m_maxEps(0),
+      m_deltaChord(0),
+      m_deltaOneStep(0),
+      m_maxStep(1. * m),
+      m_integratorStepper("NystromRK4"),
+      m_fieldComponentX(0),
+      m_fieldComponentY(0),
+      m_fieldComponentZ(-6. * tesla),
+      m_fieldRadMax(6 * m),
+      m_fieldZMax(6. * m) {
+  declareInterface<IG4MagneticFieldTool>(this);
+
+  declareProperty("FieldOn", m_fieldOn, "Switch to turn field off");
+  declareProperty("DeltaChord", m_deltaChord, "Missing distance for the chord finder");
+  declareProperty("DeltaOneStep", m_deltaOneStep, "Delta(one-step)");
+  declareProperty("MinimumEpsilon", m_minEps, "Minimum epsilon (see G4 documentation)");
+  declareProperty("MaximumEpsilon", m_maxEps, "Maximum epsilon (see G4 documentation)");
+  declareProperty("MaximumStep", m_maxStep, "Maximum step length in field (see G4 documentation)");
+
+  declareProperty("IntegratorStepper", m_integratorStepper = "NystromRK4", "Integrator stepper name");
+  declareProperty("FieldComponentX", m_fieldComponentX, "Field X component");
+  declareProperty("FieldComponentY", m_fieldComponentY, "Field Y component");
+  declareProperty("FieldComponentZ", m_fieldComponentZ, "Field Z component");
+  declareProperty("FieldRMax", m_fieldRadMax, "Field max radius");
+  declareProperty("FieldZMax", m_fieldZMax, "field max Z");
+}
 
+G4ConstantMagneticFieldTool::~G4ConstantMagneticFieldTool() {
+  if (nullptr != m_field) delete m_field;
 }
 
-//=============================================================================
-// Destructor 
-//=============================================================================
+StatusCode G4ConstantMagneticFieldTool::initialize() {
+  StatusCode sc = GaudiTool::initialize();
+  if (sc.isFailure()) return sc;
 
-G4ConstantMagneticFieldTool::~G4ConstantMagneticFieldTool()
-{
-}
+  if (m_fieldOn) {
+    m_field =
+        new sim::G4ConstantField(m_fieldComponentX, m_fieldComponentY, m_fieldComponentZ, m_fieldRadMax, m_fieldZMax);
```

Comment:
OK, I added a comment (ownership stays with the tool, the manager only has an observing pointer).